src/main/java/server/HttpServer.java

```diff
@@ -1,30 +1,45 @@
 package com.td.HttpServer;
 import java.io.IOException;
 import java.net.ServerSocket;
+
 public class HttpServer{
 
-  private static ArgumentParser arguments;
+  private ServerSocket serverSocket;
+  private HttpServerRunner httpServerRunner;
+  private CustomHandlerRouter customHandlerRouter;
+  private FileIO fileIO;
 
-  public static void main(String[] args) {
-    arguments = new ArgumentParser(args);
-    ServerSocket serverSocket;
+  public HttpServer(int portNumber, String directory) {
     try {
-      serverSocket = new ServerSocket(arguments.getPortNumber());
+      serverSocket = new ServerSocket(portNumber);
     }
     catch (IOException e) {
-      System.out.println("\nPort number " + arguments.getPortNumber() + " can not be opened. It may be in use by another application.\n");
+      System.out.println("\nPort number " + portNumber + " can not be opened. It may be in use by another application.\n");
       return;
```

Comment:
So this is a little controvesial and _probably_ want to avoid early returns from a constructor. If you return (and not blow up) then you'll have a "partially" constructed object. Now what the heck do you do with a partially constructed object?